### ISSUE TITLE
Do not clone immutable values

### ISSUE THREAD

**Issue by jonaskello** [dharmafly/jsonpatch.js#31]
When working with libraries that guarantee that values are immutable, it would be nice to have an option to not clone the value.

This could be implemented with an extra option to ```apply_patch``` that will skip the internal cloning of the value.

For example:

```js
const myImmutableValue = getImmutableValue({ bar: "foo"});
const patch = {op: "replace", path: "/foo", value: myImmutableValue};
const doc = {foo: "bar"};
const cloneValueOption = false;
const newDoc = apply_patch(doc, patch, cloneValueOption);
console.log(newDoc.foo === myImmutableValue); // true because it is the exact same object
```

**Comment by almost** [comment]
I actually think this should be the default way it works. I'm not entirely sure why I decided to clone values from the patch. I'll update it and release (with a major version change)

**Comment by almost** [comment]
It's now published as version 3.0.0

**Comment by jonaskello** [comment]
Perfect! Thanks for the fast response and release!
